### PR TITLE
Allow for lazy loading of other modules with lazy loading

### LINKED ISSUE #1960: Cannot import openff.toolkit with most recent openff-units release
**Describe the bug**

The top-level lazy-loading doesn't account for lazy loading in its dependencies itself. I think that's why the most recent openff-units release broke importing from the openff-toolkit.

**To Reproduce**

Create an environment with openff-units=0.2.3 and a recent version of the openff toolkit.

```
>>> from openff.toolkit.utils.ambertools_wrapper import AmberToolsToolkitWrapper
```

**Output**

**Computing environment (please complete the following information):**
 - Operating system
 - Output of running `conda list`

```
File "/Users/runner/micromamba/envs/openff-nagl-test/lib/python3.11/site-packages/openff/toolkit/utils/toolkit_registry.py", line 10, in <module>
    from openff.toolkit.utils.ambertools_wrapper import AmberToolsToolkitWrapper
  File "/Users/runner/micromamba/envs/openff-nagl-test/lib/python3.11/site-packages/openff/toolkit/utils/ambertools_wrapper.py", line 16, in <module>
    from openff.toolkit import Quantity, unit
  File "<frozen importlib._bootstrap>", line 1229, in _handle_fromlist
  File "/Users/runner/micromamba/envs/openff-nagl-test/lib/python3.11/site-packages/openff/toolkit/__init__.py", line 82, in __getattr__
    return mod.__dict__[name]
           ~~~~~~~~~~~~^^^^^^
KeyError: 'Quantity'
```

**Additional context**

### PR DESCRIPTION
**Pull request by lilyminium**
Fixes #1960 

- [x] Tag issue being addressed
- [ ] Add [tests](https://github.com/openforcefield/openff-toolkit/tree/main/openff/toolkit/_tests)
- [ ] Update docstrings/[documentation](https://github.com/openforcefield/openff-toolkit/tree/main/docs), if applicable
- [ ] [Lint](https://open-forcefield-toolkit.readthedocs.io/en/latest/developing.html#style-guide) codebase
- [x] Update [changelog](https://github.com/openforcefield/openff-toolkit/blob/main/docs/releasehistory.md)

### CODE REVIEW

**Comment by lilyminium** [comment]
Failures are due to mypy and I'm not sure are related to the changes here? The highlighted errors look like they're being found in different files.

**Comment by mattwthompson** [comment]
I re-kicked CI, the typing failures shouldn't reproduce with `openff-units` 0.2.2

**Comment by j-wags** [comment]
Hm, still getting [mypy errors](https://github.com/openforcefield/openff-toolkit/actions/runs/11716655045/job/32671793280?pr=1961#step:4:666), even though mypy version is [unchanged](https://github.com/openforcefield/openff-toolkit/actions/runs/11706689483/job/32604364572#step:4:668) from successful runs (1.13.0). Thoughts?

**Comment by mattwthompson** [comment]
That's surprising. I'll have a quick look

**Comment by mattwthompson** [comment]
I can't reproduce this locally, using `devtools/conda-envs/openeye.yaml`, which pulls down 0.2.0.

I think keeping that pin in place in the testing environment, with the intent to remove it sometime, makes the most sense here

File: devtools/conda-envs/openeye.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ dependencies:
   - openmm >=7.6
   - openff-forcefields >=2023.05.1
   - smirnoff99Frosst
-  - openff-units =0.2.0
+  - openff-units
```
Comment (mattwthompson) [inline devtools/conda-envs/openeye.yaml:19]:
```suggestion
  - openff-units =0.2.0
```

File: devtools/conda-envs/rdkit.yaml [hunk 2]
```diff
@@ -15,7 +15,7 @@ dependencies:
   - openmm >=7.6
   - openff-forcefields >=2023.05.1
   - smirnoff99Frosst
-  - openff-units =0.2.0
+  - openff-units
```
Comment (mattwthompson) [inline devtools/conda-envs/rdkit.yaml:18]:
```suggestion
  - openff-units =0.2.0
```

File: devtools/conda-envs/test_env.yaml [hunk 3]
```diff
@@ -15,7 +15,7 @@ dependencies:
   - openmm >=7.6
   - openff-forcefields >=2023.05.1
   - smirnoff99Frosst
-  - openff-units =0.2.0
+  - openff-units
```
Comment (mattwthompson) [inline devtools/conda-envs/test_env.yaml:18]:
```suggestion
  - openff-units =0.2.0
```

**Review by mattwthompson — COMMENTED** [review]
Re: above, these "suggestions" should just revert the pins back to what they were

**Comment by j-wags** [comment]
Just went down the rabbit hole of why we're pinned to openff-units 0.2.0 - for future archaeology, [this](https://github.com/openforcefield/openff-toolkit/pull/1646) is why. Agree with just reinstating the pin to get this PR moving.

**Review by j-wags — APPROVED** [review]
A little hesitant here because we're not able to continue test this because of the mypy issues. But this was clearly broken in production so it's worth going ahead anyway.

**Pull request merged**